streaming/logging.js

```diff
@@ -0,0 +1,98 @@
+const { pino } = require('pino');
+const { pinoHttp, stdSerializers: pinoHttpSerializers } = require('pino-http');
+const uuid = require('uuid');
+
+/**
+ * Generates the Request ID for logging and setting on responses
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} [res]
+ * @returns {import("pino-http").ReqId}
+ */
+function generateRequestId(req, res) {
+  if (req.id) {
+    return req.id;
+  }
+
+  req.id = uuid.v4();
+
+  // Allow for usage with WebSockets:
+  if (res) {
+    res.setHeader('X-Request-Id', req.id);
+  }
+
+  return req.id;
+}
+
+/**
+ * Request log sanitizer to prevent logging access tokens in URLs
+ * @param {http.IncomingMessage} req
+ */
+function sanitizeRequestLog(req) {
+  const log = pinoHttpSerializers.req(req);
+  if (typeof log.url === 'string' && log.url.includes('access_token')) {
+    log.url = log.url.replace(/(access_token)=([A-Z0-9a-z_-]+)/gi, '$1=[Redacted]');
```

Comment:
@ClearlyClaire could you confirm that access tokens conform to this regex? I believe they should, but I'm not 100% sure.

Reply:
As far as I understand Devise's code, this uses `SecureRandom.urlsafe_base64` without padding, so this looks accurate!